cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java

```diff
@@ -274,6 +280,23 @@ private static void writeAndHandleProgress(InputStream inputStream, FileOutputSt
         }
     }
 
+    private static void writeAndHandleProgressWithoutLogging(InputStream inputStream, FileOutputStream outputStream,
+                                                             String fullPkgName, LogFormatter logFormatter) {
+        int count;
+        byte[] buffer = new byte[1024];
+
+        try {
+            while ((count = inputStream.read(buffer)) > 0) {
+                outputStream.write(buffer, 0, count);
+            }
+        } catch (IOException e) {
+            PrintStream printStream = System.out;
+
+            printStream.println(logFormatter.formatLog(fullPkgName + "pulling the package from central failed"));
+
+        }
+    }
+
     /**
      * Create nightly build meta file.
      *
```

Comment:
Can we add some tests how can you disable printing when using ProjectAPI ?

Reply:
Added tests. Will add tests to distribution as well.